fixed-hash/src/hash.rs

```diff
@@ -6,735 +6,646 @@
-			}
-
-			/// Convert a slice of bytes of length `len()` to an instance of this type.
-			#[deprecated(
-				since = "0.2.5",
-				note = "unconventional API, replaced by `new_from_slice` in version 0.3"
-			)]
-			pub fn from_slice(src: &[u8]) -> Self {
-				let mut r = Self::new();
-				r.clone_from_slice(src);
-				r
+			pub fn as_ptr(&self) -> *const u8 {
+				self.as_bytes().as_ptr()
 			}
 
-			/// Copy the data of this object into some mutable slice of length `len()`.
-			#[deprecated(
-				since = "0.2.5",
-				note = "use `std::slice` API instead"
-			)]
-			pub fn copy_to(&self, dest: &mut[u8]) {
-				let min = ::core::cmp::min($size, dest.len());
-				dest[..min].copy_from_slice(&self.0[..min]);
+			/// Returns a mutable raw pointer to the value.
+			#[inline]
+			pub fn as_mut_ptr(&mut self) -> *mut u8 {
+				self.as_bytes_mut().as_mut_ptr()
+			}
+
+			/// Assign the bytes from the byte slice `src` to `self`.
+			///
+			/// # Note
+			///
+			/// The given bytes are interpreted in big endian order.
+			///
+			/// # Panics
+			///
+			/// If the length of `src` and the number of bytes in `self` do not match.
+			pub fn assign_from_slice(&mut self, src: &[u8]) {
+				$crate::core::assert_eq!(src.len(), $n_bytes);
+				self.as_bytes_mut().copy_from_slice(src);
+			}
+
+			/// Create a new fixed-hash from the given slice `src`.
+			///
+			/// # Note
+			///
+			/// The given bytes are interpreted in big endian order.
+			///
+			/// # Panics
+			///
+			/// If the length of `src` and the number of bytes in `Self` do not match.
+			pub fn from_slice(src: &[u8]) -> Self {
+				$crate::core::assert_eq!(src.len(), $n_bytes);
+				let mut ret = Self::zero();
+				ret.assign_from_slice(src);
+				ret
 			}
 
 			/// Returns `true` if all bits set in `b` are also set in `self`.
-			#[deprecated(
-				since = "0.2.5",
-				note = "will be renamed to `covers` in version 0.3"
-			)]
-			pub fn contains<'a>(&'a self, b: &'a Self) -> bool {
+			#[inline]
+			pub fn covers(&self, b: &Self) -> bool {
 				&(b & self) == b
 			}
 
 			/// Returns `true` if no bits are set.
+			#[inline]
 			pub fn is_zero(&self) -> bool {
-				self.eq(&Self::new())
-			}
-
-			/// Returns the lowest 8 bytes interpreted as a BigEndian integer.
-			#[deprecated(
-				since = "0.2.5",
-				note = "will be renamed to `low_u64_be` in version 0.3"
-			)]
-			pub fn low_u64(&self) -> u64 {
-				let mut ret = 0u64;
-				for i in 0..::core::cmp::min($size, 8) {
-					ret |= (self.0[$size - 1 - i] as u64) << (i * 8);
-				}
-				ret
+				self.as_bytes().iter().all(|&byte| byte == 0u8)
 			}
+		}
 
-			impl_std_for_hash_internals!($from, $size);
+		/// Utilizies using the `byteorder` crate.
+		#[cfg(feature = "byteorder-support")]
+		impl $name {
+			/// Returns the least significant `n` bytes as slice.
+			///
+			/// # Panics
+			///
+			/// If `n` is greater than the number of bytes in `self`.
+			#[inline]
+			fn least_significant_bytes(&self, n: usize) -> &[u8] {
```

Comment:
Docs won't be shown in default mode. However, I like documenting private parts as well since you never know when they will be published. And besides that documentation doesn't hurt for maintenance purposes, right? :)